llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -162,6 +162,10 @@ static cl::opt<bool> EmitJumpTableSizesSection(
     cl::desc("Emit a section containing jump table addresses and sizes"),
     cl::Hidden, cl::init(false));
 
+static cl::opt<bool> AnnotateJumpTables("annotate-jump-tables",
+                                        cl::desc("Annotate jump tables"),
+                                        cl::Hidden, cl::init(false));
```

Comment:
`cl::opt` is not great because it usually means that the command line flag that would trigger this needs to also be passed to the linker when performing LTO.  Instead, I think a better approach (though it's more code in LLVM) is to have a module level metadata to signal this codegen behavior, and have the front end emit that when the command line flag is set.

Reply:
OK I'll look into that. Suggestions welcome on useful examples.

Reply:
fc018ebb608ee0c1239b405460e49f1835ab6175 was exactly this.  Perhaps we can add a new `-f` flag to clang for this?

(That way we have `-fannotate-jump-tables`/`-fno-annotate-jump-tables` for the compiler, rather than `-mllvm -annotate-jump-tables` for the compiler (and the linker for LTO)).